system/Router/RouteCollection.php

```diff
@@ -781,30 +781,41 @@ public function presenter(string $name, ?array $options = null): RouteCollection
             }
         }
 
+        $as = (isset($options['as'])) ? $options['as'] : null;
+
         if (in_array('index', $methods, true)) {
+            $options['as'] = (isset($options['as'])) ? $as . '.index' : null;
```

Comment:
Too much duplicate code.
I suggest replacing it with an anonymous function.
For example like this
```php
$alias = function ($alias) use ($options) {
    if (isset($options['as']) {
        $options['as'] = "{$options['as']}.{$alias}";
    }
    
    return $options;
};

// use 
$this->get($name . '/show/' . $id, $newName . '::show/$1', $alias('show'));
```
Using an anonymous function:
- code is being optimized
- the original ``$options`` array will not be changed.